### ISSUE TITLE
`main` entry points to ESM, not CommonJS

### ISSUE THREAD

**Issue by brownieboy** [josdejong/immutable-json-patch#2]
In the main package.json, you have the `main` and `module` entries both pointing to the same ESM folder:

```json
  "main": "lib/esm/index.js",
  "module": "lib/esm/index.js",
  "browser": "lib/umd/immutableJSONPatch.js",
  "types": "lib/types/index.d.ts",
  "exports": {
    ".": {
      "import": "./lib/esm/index.js",
      "require": "./lib/umd/immutableJSONPatch.js",
      "browser": "./lib/umd/immutableJSONPatch.js",
      "types": "./lib/types/index.d.ts"
    }
  },
```

You're outputting a /lib/cjs folder, so the `main` entry should point to that one, shouldn't it?

I'm working with an older version of Jest, which is ignoring your new `exports` entry.   To make it work, I have to change the `main` entry manually, like so:

```json
  "main": "lib/cjs/index.js",
```

**Comment by josdejong** [comment]
Hm, yeah you may be right. I was assuming that you can let it point to an ESM file since the package defines `"type": "module"`. I did a quick search but cannot find clear documentation about that. Do you know of some clear documentation about the `main` field and related fields?

**Comment by brownieboy** [comment]
Thanks for your quick response, @josdejong .

I've subsequently updated my Jest to a version that correctly honours the `exports` field, so this is no longer an issue for me.

To answer your question though, I found this doco on the NodeJS site: https://nodejs.org/api/packages.html#dual-commonjses-module-packages, which says:

> Prior to the introduction of support for ES modules in Node.js, it was a common pattern for package authors to include both CommonJS and ES module JavaScript sources in their package, with package.json ["main"](https://nodejs.org/api/packages.html#main) specifying the CommonJS entry point and package.json "module" specifying the ES module entry point. This enabled Node.js to run the CommonJS entry point while build tools such as bundlers used the ES module entry point, since Node.js ignored (and still ignores) the top-level "module" field.

So "a common pattern" isn't actually a definitive spec that says this is how it's supposed to work, but I've seen a lot of packages that do it this way.

**Comment by josdejong** [comment]
Good to hear you've found a working solution!

Thanks for digging this up. It makes sense, I indeed have seen `main` used for CommonJS normally. I'll change the `main` field to adhere to this common pattern.

**Comment by josdejong** [comment]
Fixed now in `v5.1.3`
